### PR TITLE
fix: center the widget when height is min auto height

### LINKED ISSUE #20119: [Bug]: Widgets are not aligned by default
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Widgets which have auto height enabled are no longer vertically aligned by default with one another. This is a break in the UI building experience brought about by the Auto height feature.

We need to make widgets aligned by default.

### Steps To Reproduce

1. Add a button widget
2. Add a text widget right beside the button widget
3. See the alignment issue.

https://user-images.githubusercontent.com/103687/214925712-7e606476-d0fe-4ff4-8d95-c39927c09d29.mov

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud and Self hosted

### PR DESCRIPTION
**Pull request by ankurrsinghal**
## Description
Widgets like Text widget cannot be centered vertically by default in the canvas as they used to before the Auto Height feature. The reason for this is that the Auto height feature allows components to specify the height of the widgets, instead of the other way around (which used to be the case before auto height)

To fix this, we've made it so that, when a widget -- which has auto height enabled -- has a height equal to the minimum height and is set to the minimum possible height (4 rows), the contents in the component are centered vertically.

Fixes #20119 

#### Before

https://user-images.githubusercontent.com/103687/214924485-b6d7f1b8-466f-4e3e-8468-213bab5ca3cd.mov

#### After

https://user-images.githubusercontent.com/103687/214924796-d9e5b854-8e59-47dd-bee0-a78719ef5dbb.mov

## Type of change
- Breaking change. This changes the behaviour of widgets.

## How Has This Been Tested?
- Manual: See Test Plan for details
- Cypress: We'll add a test to see if widgets added to the canvas have their components centered.

### Test Plan
https://github.com/appsmithorg/TestSmith/issues/2172

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [NA] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [PENDING] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [NA] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by dilippitchika — DISMISSED** [review]

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_Content_Alignment_spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
```
Comment (riodeuno) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_Content_Alignment_spec.ts:1]: @NandanAnantharamu @Aishwarya-U-R Please review this PR.

**Review by marks0351 — DISMISSED** [review]

**Review by marks0351 — APPROVED** [review]

**Comment by Sripriya93** [comment]
@riodeuno This PR is good to go.

**Review by marks0351 — APPROVED** [review]

**Pull request merged**